erpnext/selling/doctype/customer/customer.js

```diff
@@ -148,7 +173,9 @@ frappe.ui.form.on("Customer", {
 	},
 	validate: function(frm) {
 		if(frm.doc.lead_name) frappe.model.clear_doc("Lead", frm.doc.lead_name);
-
+		if(frm.days_selected) {
+			frm.set_value("delivery_days", frm.days_selected.get_value().toString())
```

Comment:
Don't change this to a string, keep it as an array

```suggestion
			frm.set_value("delivery_days", frm.days_selected.get_value())
```